### PR TITLE
Feature/time manager

### PR DESCRIPTION
**Pull request by wbeltran777**
Important methods here are BeginTimer and CancelTimer
BeginTimer takes in a string, the time you want the timer to take, and the method you want to fire when the time ends. CancelTimer interrupts whatever method you pass it.

### CODE REVIEW

File: Assets/Script/Game/ParkingDetection.cs [hunk 1]
```diff
@@ -99,13 +104,35 @@ private void OnTriggerStay2D(Collider2D collision)
     {
         if (IsParked(collision))
         {
-            // Relies on the collider to be 2 levels down
-            ScoreManager.Instance.CarScored(collision.transform.parent.gameObject);
-            TargetManager.ActivateRandomParkingSpot();
-            UnchooseParkingSpot();
+            if(!m_carIsInParkingSpot)
```
Comment (m4rtint) [inline Assets/Script/Game/ParkingDetection.cs]: Better to put this boolean inside the "IsParked" method.

File: Assets/Script/Managers/TimeManager.cs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        {
+            m_instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+        }
+        WorldManager.OnFixedUpdate += TimeManagerUpdate;
+        m_timerEventsAvailable = new Dictionary<string, TimeDelegate>();
+        m_TimerEnd1 += DummyMethodForEvents;
```
Comment (m4rtint) [inline Assets/Script/Managers/TimeManager.cs]: Remember to have a place to remove the delegates

**Comment by mikeslgoh** [comment]
[Musical Chairs: TimeManager](https://trello.com/c/OibqMk1j/77-musical-chairs-timemanager)

File: Assets/Script/Game/ParkingDetection.cs [hunk 3]
```diff
@@ -18,6 +18,10 @@ public class ParkingDetection : MonoBehaviour
 
     private bool m_isChosenToBeParkingSpot = false;
 
+    private bool m_carIsInParkingSpot = false;
+
+    private int testVarDeleteLater = 0;
```
Comment (m4rtint) [inline Assets/Script/Game/ParkingDetection.cs]: delete?

**Comment by m4rtint** [comment]
It doesn't seem to work even after I pull

Did you save the scene? or save the project?

**Comment by wbeltran777** [comment]
Recloned and tested it again, lemme know if it works now

File: Assets/Script/Game/Car/CarController.cs [hunk 4]
```diff
@@ -57,6 +61,15 @@ public void SetIndicatorTarget(Transform target)
         m_indicator.Target = target;
     }
 
+    /// <summary>
+    /// Set amount of time car must be in parking spot
+    /// </summary>
+    /// <param name="parkingDuration"> time car must remain in parking spot </param>
+    public void SetParkingTimerAmount(float parkingDuration)
```
Comment (m4rtint) [inline Assets/Script/Game/Car/CarController.cs]: Why does this have to be public? I don't see it used anywhere else.

File: Assets/Script/Managers/PlayerManager.cs [hunk 5]
```diff
@@ -79,6 +79,7 @@ public void Initialize()
         }
 
         CheckNumberOfPlayers();
+        //ScoreManager.CarScoredEvent += ResetParkingTimers;
```
Comment (m4rtint) [inline Assets/Script/Managers/PlayerManager.cs]: delete

File: Assets/Script/Managers/PlayerManager.cs [hunk 6]
```diff
@@ -98,6 +99,17 @@ private void Awake()
...
     }
 
+    /// <summary>
+    /// Iterate over every car and set their internal clock back to 0 
+    /// </summary>
+    public void ResetParkingTimers()
+    {
+        foreach (PlayerController car in m_players)
+        {
+            car?.GetComponentInChildren<ParkingTimer>().ResetParkingWheel();
```
Comment (m4rtint) [inline Assets/Script/Managers/PlayerManager.cs]: Just an fyi - 
The way GetComponentInChildren works, is that it loops through every child object and look through every component until it finds the correct one.

**Pull request merged**